### ISSUE TITLE
Why memory addresses that belong to different processes are the same ?

### ISSUE THREAD

**Issue by BoyuanYan** [openai/baselines#304]
I've tried command ``python -m baselines.a2c.run_atari``. And when I inserted one line ``print("env's address is : {}".format(id(env)))`` before [subproc_vec_env.py](https://github.com/openai/baselines/blob/master/baselines/common/vec_env/subproc_vec_env.py#L8), I got the output:
>Logging to /tmp/openai-2018-03-01-00-15-00-194270
env's address is : 140342766730992
env's address is : 140342766726896
env's address is : 140342766730992
env's address is : 140342766735088
2018-03-01 00:15:01.063327: I tensorflow/core/platform/cpu_feature_guard.cc:137] Your CPU supports instructions that this TensorFlow binary was not compiled to use: SSE4.1 SSE4.2 AVX AVX2 FMA
env's address is : 140342766730992
env's address is : 140342766730992
env's address is : 140342766730992
env's address is : 140342766730992
env's address is : 140342766726896
env's address is : 140342766730992
env's address is : 140342766730992
env's address is : 140342766730992
env's address is : 140342766726896
env's address is : 140342766730992
env's address is : 140342766730992
env's address is : 140342766730992

So I wonder why the memory addresses from different processes are the same. does A2C runs same games with same condition in parallel?

**Comment by ghostFaceKillah** [comment]
No, games in A2C in general are run with different state in parallel.

It depends what you mean by inserting print before subproc_vec_env.py, but what could be confusing you is the fact that it is possible for different processes to internally point to the same memory address and see different things. This is because of the virtual memory mechanism implemented by operating system and hardware - each process has it's own virtual address space. It can happen that two same virtual addresses in two separate processes point to different places in physical memory.
